'talosctl cluster show' panics

## Bug Report

### Description

```
$ talosctl cluster show
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x68 pc=0x203b69a]

goroutine 1 [running]:
github.com/siderolabs/talos/pkg/provision/providers/docker.(*provisioner).Reflect(0x31e8838?, {0x31e8838, 0xc0007e7140}, {0x2bfe2e6, 0xd}, {0xc000076400?, 0xc0007dbb60?})
	/src/pkg/provision/providers/docker/reflect.go:70 +0x4ba
github.com/siderolabs/talos/cmd/talosctl/cmd/mgmt/cluster.show({0x31e8838, 0xc0007e7140})
	/src/cmd/talosctl/cmd/mgmt/cluster/show.go:44 +0xc6
github.com/siderolabs/talos/pkg/cli.WithContext({0x31e8870?, 0xc000050098?}, 0x2d703a0)
	/src/pkg/cli/context.go:40 +0x1d1
github.com/siderolabs/talos/cmd/talosctl/cmd/mgmt/cluster.glob..func3(0x49caa60?, {0x4f79288?, 0x0?, 0x0?})
	/src/cmd/talosctl/cmd/mgmt/cluster/show.go:32 +0x2e
github.com/spf13/cobra.(*Command).execute(0x49caa60, {0x4f79288, 0x0, 0x0})
	/.cache/mod/github.com/spf13/cobra@v1.6.1/command.go:916 +0x862
github.com/spf13/cobra.(*Command).ExecuteC(0x49c8da0)
	/.cache/mod/github.com/spf13/cobra@v1.6.1/command.go:1044 +0x3bd
github.com/siderolabs/talos/cmd/talosctl/cmd.Execute()
	/src/cmd/talosctl/cmd/root.go:51 +0x37f
main.main()
	/src/cmd/talosctl/main.go:19 +0x25
```

### Logs

### Environment

- Talos version:
 $ talosctl version --client
Client:
	Tag:         v1.3.3
	SHA:         c2cdf54a
	Built:
	Go version:  go1.19.4
	OS/Arch:     linux/amd64
- Platform: Ubuntu 22.04.1 LTS

--- a/pkg/provision/providers/docker/reflect.go
+++ b/pkg/provision/providers/docker/reflect.go
@@ -14,6 +14,7 @@ import (
 	"github.com/siderolabs/talos/pkg/provision"
 )
 
+//nolint:gocyclo
 func (p *provisioner) Reflect(ctx context.Context, clusterName, stateDirectory string) (provision.Cluster, error) {
 	res := &result{
 		clusterInfo: provision.ClusterInfo{
@@ -67,9 +68,15 @@ func (p *provisioner) Reflect(ctx context.Context, clusterName, stateDirectory s
 			return nil, err
 		}
 
-		addr, err := netip.ParseAddr(node.NetworkSettings.Networks[res.clusterInfo.Network.Name].IPAddress)
-		if err != nil {
-			return nil, err
+		var ips []netip.Addr
+
+		if network, ok := node.NetworkSettings.Networks[res.clusterInfo.Network.Name]; ok {
+			addr, err := netip.ParseAddr(network.IPAddress)
+			if err != nil {
+				return nil, err
+			}
+
+			ips = append(ips, addr)
 		}
 
 		res.clusterInfo.Nodes = append(res.clusterInfo.Nodes,
@@ -78,7 +85,7 @@ func (p *provisioner) Reflect(ctx context.Context, clusterName, stateDirectory s
 				Name: strings.TrimLeft(node.Names[0], "/"),
 				Type: t,
 
-				IPs: []netip.Addr{addr},
+				IPs: ips,
 			})
 	}
 

